File: 3811-reverse-degree-of-a-string/3811-reverse-degree-of-a-string.java
class Solution {
    private int getReverseDegree(char c) {
        return 26 - (c - 'a');
    }

    public int reverseDegree(String s) {
        int result = 0;
        char[] chars = s.toCharArray();

        for (int i = 0; i < chars.length; i++) {
            result += getReverseDegree(chars[i]) * (i + 1);
        }

        return result;
    }
}

File: 3811-reverse-degree-of-a-string/3811-reverse-degree-of-a-string.kt
class Solution {
    private inline fun Char.getReverseDegree(): Int = 26 - (this - 'a')

    fun reverseDegree(s: String): Int {
        var result = 0
        s.forEachIndexed { i, c ->
            result += c.getReverseDegree() * (i + 1)
        }

        return result
    }
}